### DISCUSSION TITLE
Anyone can help with this ?

### DISCUSSION THREAD

**Discussion by robyorke95** [community/community#122662] [Programming Help]
ruby coding environment

### Body

i am following this guide to prepare my coding enviromnment https://github.com/lewagon/setup/blob/master/windows.md
i am using windows 11 everything went smooth untill the rbnev part when i use the following gitclone 
git clone https://github.com/rbenv/ruby-build.git ~/.rbenv/plugins/ruby-build i get this error
Cloning into '/home/rov/.rbenv/plugins/ruby-build'...
error: RPC failed; HTTP 400 curl 22 The requested URL returned error: 400
fatal: error reading section header 'shallow-info'

i cannot seem to be able to figure out how to fix this

**Comment by CopperEagle** [comment]
Hey @robyorke95 

I could not reproduce the error. Guessing based on the "shallow-info" error, how about you try this:

1) A quick and cheap jab is to try `git clone --depth 20  https://github.com/rbenv/ruby-build.git ~/.rbenv/plugins/ruby-build`. Pretty sure that won't fix the problem though, as the repository is not that deep.

2) Make sure  `git` is up to date. There was this issue sometimes issues in older versions of git. Also make sure there is no firewall or vpn settings that filter some traffic. [Here](https://stackoverflow.com/questions/65069443/git-cannot-clone-or-pull-repository-error-output) is more information that could potentially help you.

3) The **easiest** way, if you don't need the repositories' [history](https://stackoverflow.com/questions/33857243/difference-between-clone-in-desktop-and-zip-download), is to do it like this:
- Download the [zip](https://github.com/rbenv/ruby-build/archive/refs/heads/master.zip) from the repository.
- Extract to where you want to have it, navigate in the terminal to the repository
- Call `git init`.

4) You could consider using the Git*hub* commandline tool, `gh`. Then use 
```bash
gh repo clone rbenv/ruby-build
```

**Reply by CopperEagle** [reply]
So @robyorke95 were you able to fix the issue? If you did, consider sharing your solution with the community or marking an answer so others in the future see what works. :smiley:

**Comment by aroonmircam** [comment]
The error you are seeing suggests that there is a problem with the URL you are using to clone the ruby-build repository into your .rbenv/plugins/ directory. This could be due to a number of reasons, such as connection issues, problems with the URL provided, or problems with your git configuration.

1. Verify the URL: Make sure the URL you are using to clone the ruby-build repository is correct and accessible. You can check the URL in the browser to make sure it works correctly.

2. Cloning without depth: Try to clone the repository without the --depth option, which means that you will clone the entire history of the repository instead of just the latest version. You can do this by running the following command: 

`git clone https://github.com/rbenv/ruby-build.git ~/.rbenv/plugins/ruby-build`
